installer/default_platform.conf

```diff
@@ -1,119 +1,5 @@
 #!/bin/sh
 
-#  Copyright (C) 2014-2015 Curt Brune <curt@cumulusnetworks.com>
-#  Copyright (C) 2014-2015 david_yang <david_yang@accton.com>
-#
-#  SPDX-License-Identifier:     GPL-2.0
-
-# Appends a command to a trap, which is needed because default trap behavior is to replace
-# previous trap for the same signal
-# - 1st arg:  code to add
-# - ref: http://stackoverflow.com/questions/3338030/multiple-bash-traps-for-the-same-signal
-_trap_push() {
-    local next="$1"
-    eval "trap_push() {
-        local oldcmd='$(echo "$next" | sed -e s/\'/\'\\\\\'\'/g)'
-        local newcmd=\"\$1; \$oldcmd\"
-        trap -- \"\$newcmd\" EXIT INT TERM HUP
-        _trap_push \"\$newcmd\"
-    }"
-}
-_trap_push true
-
-read_conf_file() {
-    local conf_file=$1
-    while IFS='=' read -r var value || [ -n "$var" ]
-    do
-        # remove newline character
-        var=$(echo $var | tr -d '\r\n')
-        value=$(echo $value | tr -d '\r\n')
-        # remove comment string
-        var=${var%#*}
-        value=${value%#*}
-        # skip blank line
-        [ -z "$var" ] && continue
-        # remove double quote in the beginning
-        tmp_val=${value#\"}
-        # remove double quote in the end
-        value=${tmp_val%\"}
-        eval "$var=\"$value\""
-    done < "$conf_file"
-}
-
-# Main
-set -e
-cd $(dirname $0)
-
-if [ -d "/etc/sonic" ]; then
-    echo "Installing SONiC in SONiC"
-    install_env="sonic"
-elif grep -Fxqs "DISTRIB_ID=onie" /etc/lsb-release > /dev/null
-then
-    echo "Installing SONiC in ONIE"
-    install_env="onie"
-else
-    echo "Installing SONiC in BUILD"
-    install_env="build"
-fi
-
-if [ -r ./machine.conf ]; then
-    read_conf_file "./machine.conf"
-fi
-
-if [ -r ./onie-image.conf ]; then
-. ./onie-image.conf
-fi
-
-echo "ONIE Installer: platform: $platform"
-
-# Make sure run as root or under 'sudo'
-if [ $(id -u) -ne 0 ]
-    then echo "Please run as root"
-    exit 1
-fi
-
-# get running machine from conf file
-if [ -r /etc/machine.conf ]; then
-    read_conf_file "/etc/machine.conf"
-elif [ -r /host/machine.conf ]; then
-    read_conf_file "/host/machine.conf"
-elif [ "$install_env" != "build" ]; then
-    echo "cannot find machine.conf"
-    exit 1
-fi
-
-echo "onie_platform: $onie_platform"
-
-# Get platform specific linux kernel command line arguments
-ONIE_PLATFORM_EXTRA_CMDLINE_LINUX=""
-
-# Default var/log device size in MB
-VAR_LOG_SIZE=4096
-
-[ -r platforms/$onie_platform ] && . platforms/$onie_platform
-
-# Verify image platform is inside devices list
-if [ "$install_env" = "onie" ]; then
```

Comment:
Hi, after removing this part, how do you handle the below cases:
1. install sonic-mellnoax.bin in broadcom ASIC. (That could be caused by human error)
2. install sonic-broadcom.bin in mellonox ASIC.

Reply:
Hi @liat-grozovik, I think we are on the same page. I meant we need to prevent the above cases.
Those cases were prevented by the removed code. I am asking how to prevent these cases without the below handling.
```
if ! grep -Fxq "$onie_platform" platforms_asic; then
        echo "The image you're trying to install is of a different ASIC type as the running platform's ASIC"
```

Reply:
The code you show me as removing was relocated here
https://github.com/alexrallen/sonic-buildimage/blob/sonic_arm_grub/installer/install.sh#L99

git did not handle this refactor well, apologies for that. Let me know any other questions.